src/django/api/views.py

```diff
@@ -1879,6 +1880,8 @@ def merge(self, request):
         # be made in the `facility_history.py` module's
         # `create_facility_history_dictionary` function
         merge.changeReason = 'Merged with {}'.format(target.id)
+
+        FacilityIndex.objects.get(id=merge.id).delete()
```

Comment:
When testing the extended field indexing, I discovered that FacilityIndex objects weren't being removed after a merge (when the related Facility is deleted). Added a line to remove the stale Index objects